_posts/2024-03-06-bitcoin-inquisition-#39.md

```diff
@@ -6,34 +6,57 @@ link: https://github.com/bitcoin-inquisition/bitcoin/pull/39
 permalink: /bitcoin-inquisition-39
 authors: [0xBEEFCAF3, EthanHeilman]
 components: ["consensus"]
-host: EthanHeilman
+host: EthanHeilman, arminsdev aka 0xBEEFCAF3
 status: upcoming
 commit:
 ---
 
 _Notes and questions to follow soon!_
 
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+The following pull request (PR) reinstates the usage of OP_CAT in accordance with the specifications outlined in the draft BIP, which can be found [here](https://github.com/bitcoin/bips/pull/1525). Reviewers are encouraged to familiarize themselves with the BIP before examining the code.
 
+- The primary objective of this PR is to re-enable OP_CAT utilizing OP_SUCCESS semantics, replacing OP_SUCCESS126.
+  - OP_CAT is designed to concatenate two elements on the stack.
+  - If the resulting element is smaller than the `MAX_SCRIPT_ELEMENT_SIZE`, it is placed on the stack.
+- OP_CAT should fail if there are less than two values on the stack or if a concatenated value would have a combined size greater than `MAX_SCRIPT_ELEMENT_SIZE`.
+- The PR also introduces activation parameters for the Bitcoin Signet network, aligning with the [BIN-2024-0002](https://github.com/bitcoin-inquisition/binana/blob/master/2024/BIN-2024-0002.md) standard.
+- Additionally, this PR introduces new semantics for testing taproot-related script tests.
 
 ## Questions
 
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+1. OP_CAT is defined as 0x7e. Even after replacing an OP_SUCCESS opcode, libraries can continue to use 0x7e to represent concatenation operations. Why is this the case?
+
+1. A new script verify flag is introduced: `SCRIPT_VERIFY_OP_CAT`. Is this considered a consensus-validating flag or a policy?
+
+1. If it is policy, where does the mempool policy reject transactions that use OP_CAT? Why does this policy exist?
+
+1. In `deploymentinfo.cpp`, there are both an `OP_CAT` flag and a `DISCOURAGE_OP_CAT` flag. What is the rationale behind having both of these?
+
+1. When does consensus consider OP_SUCCESS126 replaced by OP_CAT?
+
+1. Why is it important to verify if OP_CAT is being executed in a non-segwitv0 or base-script context at L474 of `interpreter.cpp` rather than inside the opcode definition?
+
+1. This PR introduces new semantics for taproot-related script tests in `script_tests.json`. What issues or inefficiencies existed with the previous testing strategy?
+
+1. Are there any additional test cases you would like to see implemented that are not covered by the functional tests or the script tests in `script_tests.json`?
```

Comment:
I'd recommend adding links to places in the code (from https://github.com/bitcoin-core-review-club/bitcoin/tree/prbitcoin-inquisition-39), as that can be a hint for how to answer the question and encourage people to do more code review. For example, you can link where `SCRIPT_VERIFY_OP_CAT` is defined/used.